routes.jsx

```diff
@@ -15,21 +16,38 @@ import Navbar from './components/navbar/navbar.jsx';
 import Footer from './components/footer/footer.jsx';
 
 const App = React.createClass({
+  getInitialState() {
+    return {
+      suppressSplashScreen: localstorage.getItem(`suppressSplashScreen`) === `true`
+    };
+  },
   componentDidMount() {
-    setTimeout(function() {
-      this.splash.classList.add(`dismissed`);
-      document.querySelector(`#app`).classList.add(`splash-dismissed`);
-    }, 3000);
+    if (this.state.suppressSplashScreen !== `true`) {
+      setTimeout(function() {
+        this.splash.classList.add(`dismissed`);
+        document.querySelector(`#app`).classList.add(`splash-dismissed`);
+      }, 3000);
+      localstorage.setItem(`suppressSplashScreen`, `true`);
```

Comment:
I'd move this line into the above function, so we make sure to skip the splash after the user has seen the splash for the full 3 seconds at least once. And mostly because we're working in React, I'm also going to say "let's not define functions on the fly. This can be a `dismissSplash` function on the React component itself, which we call with `setTimeout(this.dismissSplash, this.props.dismissTimeout)` (with a `defaultProps` that sets `dismissTimeout: 3000`)